libvast/vast/system/component_registry.hpp

```diff
@@ -47,12 +47,21 @@ class component_registry {
   /// @param label The unique label of *comp*
   /// @returns `true` if *comp* was added successfully and `false` if
   ///          an actor for *label* exists already.
-  bool add(caf::actor comp, std::string type, std::string label = {});
+  /// @pre `comp && !type.empty()`
+  [[nodiscard]] bool
+  add(caf::actor comp, std::string type, std::string label = {});
 
   /// Removes a component from the registry.
-  /// @param comp The component to remove
-  /// @returns `true` iff the component was deleted successfully.
-  bool remove(const caf::actor& comp);
+  /// @param label The label of the component.
+  /// @returns The deleted component or a default-constructed component if
+  /// *label* does not identify an existing component.
+  [[nodiscard]] component remove(const std::string& label);
```

Comment:
API wise I think both remove functions should return a `caf::expected<component>` instead. Relying on special values makes it hard to reason about the code at the call site without having read the comment above this function.

Reply:
Okay, how do you feel about making them return `caf::no_error` in case nothing is found? I.e., a default-constructed `caf::error`.

Reply:
Or `ec::lookup_error`? It's probably an error if I tried to remove a component but it's not in the registry.